src/target_info.rs

```diff
@@ -85,34 +84,52 @@ fn extract_stack_info(
         let section_range = lowest_address..=highest_address;
         let name = section.name().unwrap_or("<unknown>");
 
-        if active_ram_region.range.contains(&highest_address) {
-            log::debug!(
-                "section `{}` is in RAM at {:#010X} ..= {:#010X}",
-                name,
-                lowest_address,
-                highest_address,
-            );
+        if ram_range.contains(section_range.end()) {
+            log::debug!("section `{}` is in RAM at {:#010X?}", name, section_range);
 
-            if section_range.contains(&(initial_stack_pointer - 1)) {
+            if section_range.contains(&stack_end) {
                 log::debug!(
                     "initial SP is in section `{}`, cannot determine valid stack range",
                     name
                 );
                 return None;
-            }
-
-            if initial_stack_pointer > lowest_address && *stack_range.start() <= highest_address {
-                stack_range = highest_address + 1..=initial_stack_pointer;
+            } else if is_superset(&(stack_start..=stack_end), &section_range) {
```

Comment:
I think this doesn't match what the original code was trying to do. We don't really care about whether one range contains the other, we want to know if the current `stack_start` needs to be adjusted because of `section_range`, so the code should check if `stack_start` is in the section range, and adjust that accordingly.

Reply:
> so the code should check if `stack_start` is in the section range

Don't we just need to check if the current `stack_start` is below `section_range.end()`?

Reply:
Ah, yeah, it doesn't have to be *inside* the section